### PR TITLE
Add extra options to map and fix bugs

### PR DESCRIPTION
**Pull request by AndreJoaquim**
@binhums sorry to have this PR here stuck for so long time. We've added some features (mapbox-circle, mabox-layer, mabox-geojson, etc). And did some improvements on the map component (allowing more customisation options). We've updated mapbox to the latest version.
I've also added tests to all this features, including the MarkerCluster that already existed and hadn't any tests.

I think it is now ready for you to review, hope it is what you expect and that we can manage to create a version out of this as I think it has some nice improvements :)

Thank you!

### CODE REVIEW

File: addon/components/mapbox-markercluster.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ export default Ember.Component.extend({
   spiderLegPolylineOptions: { weight: 1.5, color: '#222', opacity: 0.5 },
   spiderfyDistanceMultiplier: 1,
   maxClusterRadius: 80,
+  hasEvents: true,
```
Comment (montoias) [inline addon/components/mapbox-markercluster.js]: To keep BC, make it by default false. 

Otherwise will have that bind

**Comment by asantos00** [comment]
@binhums sorry to have this PR here stuck for so long time. We've added some features (mapbox-circle, mabox-layer, mabox-geojson, etc). And did some improvements on the map component (allowing more customisation options).

I've also added tests to all this features, including the `MarkerCluster` that already existed and hadn't any tests.

I think it is now ready for you to review, hope it is what you expect :)

**Comment by sortins** [comment]
👍

**Comment by paulomcg** [comment]
👍

**Comment by migueldemoura** [comment]
I don't know whether I should laugh or cry at this. 😞

**Comment by asantos00** [comment]
👍

**Comment by asantos00** [comment]
😢

**Comment by paulomcg** [comment]
😭

**Pull request closed without merging**